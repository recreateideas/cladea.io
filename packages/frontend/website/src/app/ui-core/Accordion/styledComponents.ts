import styled, { css } from 'styled-components';
import {
    Accordion as MUIAccordion,
    AccordionDetails as MUIAccordionDetails,
    AccordionSummary as MUIAccordionSummary,
} from '@mui/material';

export const Container = styled.div``;
export const Accordion = styled(MUIAccordion)`
    ${(props) => {
        const {
            theme: {
                dsl: { palette },
            },
        } = props;
        return css`
            background-color: ${palette.neutrals.secondaryBg};
            box-shadow: none;
            background-image: none;
        `;
    }}
`;
export const AccordionDetails = styled(MUIAccordionDetails)`
    ${(props) => {
        const {
            theme: {
                dsl: {
                    layout: { padding },
                },
            },
        } = props;
        return css`
            padding: 0px ${padding.small} ${padding.small} ${padding.small};
        `;
    }}
`;
export const AccordionSummary = styled(MUIAccordionSummary)`
    ${(props) => {
        const {
            theme: {
                dsl: {
                    sizes,
                    layout: { padding },
                },
            },
        } = props;
        return css`
            &.Mui-expanded {
                min-height: ${sizes.accordion.medium};
            }
            > div {
                margin: ${padding.small} 0 !important;
            }
            padding: 0px ${padding.small};
        `;
    }}
`;
export const Title = styled.div``;
