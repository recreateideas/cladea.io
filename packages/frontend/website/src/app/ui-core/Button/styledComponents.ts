import styled, { css } from 'styled-components';
import { Button } from '@mui/material';

export const StyledButton = styled(Button)`
    ${(props) => {
        const {
            theme: {
                dsl: { palette },
            },
        } = props;
        return css`
            &.MuiButton-containedPrimary .MuiButton-label {
                color: ${palette.fonts.primaryInverted};
            }
        `;
    }}
`;
