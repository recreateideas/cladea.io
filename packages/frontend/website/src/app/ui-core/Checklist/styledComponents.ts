import styled from 'styled-components';

export const Container = styled.div`
    width: 100%;
    ul {
        background-color: inherit;
        .MuiListItemIcon-root {
            min-width: auto;
        }
        .MuiButtonBase-root {
        }
    }
`;
