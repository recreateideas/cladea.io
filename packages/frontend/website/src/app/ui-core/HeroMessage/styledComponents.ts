import styled, { css } from 'styled-components';

export const Container = styled.div<{ type: 'error' | 'warn' | 'success' }>`
    ${(props) => {
        const {
            theme: {
                dsl: { hexToRgba, styles, layout, palette },
            },
            type,
        } = props;
        const colorMap: {
            [type: string]: {
                color: string;
                backgroundColor: string;
            };
        } = {
            error: {
                color: palette.fonts.error,
                backgroundColor: hexToRgba(palette.secondary.orange[500], 10),
            },
            success: {
                color: palette.fonts.success,
                backgroundColor: hexToRgba(palette.secondary.green[500], 10),
            },
            warn: {
                color: palette.fonts.warn,
                backgroundColor: hexToRgba(palette.secondary.yellow[500], 10),
            },
        };
        return css`
            ${colorMap[type]}
            white-space: pre;
            padding: ${layout.padding.small};
            border-radius: ${styles.borderRadius};
            overflow-x: auto;
            .section {
                display: flex;
                .icon {
                    margin-right: ${layout.margin.small};
                }
            }
        `;
    }}
`;
