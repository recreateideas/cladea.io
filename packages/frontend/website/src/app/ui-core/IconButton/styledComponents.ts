import styled, { css } from 'styled-components';

export const Container = styled.div<{ shape?: 'square'; disabled?: boolean }>`
    ${(props) => {
        const {
            theme: {
                dsl: { palette, styles },
            },
            disabled,
            shape,
        } = props;
        const bordersRadius = shape === 'square' ? styles.borderRadius : undefined;
        return css`
            ${bordersRadius &&
            css`
                .MuiButtonBase-root {
                    border-radius: ${bordersRadius};
                }
            `}
            ${disabled &&
            css`
                &,
                * {
                    cursor: no-drop;
                    svg {
                        fill: ${palette.neutrals.inactive} !important;
                    }
                }
            `}
        `;
    }}
`;
