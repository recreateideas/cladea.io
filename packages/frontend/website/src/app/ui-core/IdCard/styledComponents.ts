import styled, { css } from "styled-components";
import { avatar } from "../images";

export const Container = styled.div<{ breakpoint: string }>`
  ${(props) => {
    const {
      theme: {
        dsl: { layout, palette },
      },
      breakpoint,
    } = props;
    const isSmall = breakpoint === "sm";
    const width = isSmall ? "100%" : "30vw";
    return css`
      border-radius: 12px;
      min-width: ${width};
      background-color: ${palette.neutrals.quaternaryBg};
      border: solid 1px ${palette.neutrals.borders};
      overflow: hidden;
      .background {
        width: 100%;
        height: 60vh;
        fill: ${palette.neutrals.secondaryBg};
      }
      z-index: ${layout.namedZIndex.leftSidebar - 1};
      .bio {
        display: flex;
        flex-direction: column;
      }
      .avatar {
        position: relative;
        border-radius: 100%;
        overflow: hidden;
        background: ${palette.neutrals.primaryBg};
        width: 20vh;
        height: 20vh;
        margin: 24px auto;
        .image-container {
          border-radius: 100%;
          overflow: hidden;
          width: inherit;
          height: inherit;
          background-image: url(${avatar});
          background-position: center;
          background-repeat: no-repeat;
          background-size: cover;
        }
        .space {
          background: unset !important;
        }
      }
      .name {
        display: flex;
        letter-spacing: 1px;
        justify-content: center;
        gap: 16px;
        .first {
        }
        .last {
          color: ${palette.tertiary.pink[500]};
        }
      }
      .position {
        width: 100%;
        .title {
          color: ${palette.secondary.neon[500]};
          text-align: center;
        }
      }
      .contacts {
        display: flex;
        flex-direction: column;
        gap: 16px;
        margin: ${isSmall ? "2vh" : "24px"} 24px;
        padding: ${isSmall ? "2vh" : "24px"};
        border: solid 1px ${palette.neutrals.borders};
        border-radius: 8px;
        background-color: ${palette.neutrals.secondaryBg};
        .contact-container {
          width: 100%;
          /* height: 40px; */
          display: flex;
          justify-content: center;
          .href-container {
            text-decoration: none;
            display: flex;
            gap: 16px;
            color: ${palette.fonts.primary};
            cursor: pointer;
            img {
              width: 36px;
            }
            .href-text {
              display: flex;
              flex-direction: column;
              justify-content: center;
              width: 80px;
            }
          }
        }
      }
    `;
  }};
`;

export const ToastContainer = styled.div`
  ${(props) => {
    const {
      theme: {
        dsl: { palette },
      },
    } = props;
    return css`
      .toast {
        display: block;
        position: fixed;
        top: 50%;
        left: 50%;
        transform: translateX(-50%);
        min-width: 450px;
        background-color: ${palette.neutrals.quaternaryBg};
        box-shadow: ${palette.neutrals.boxShadow(palette.secondary.neon[500])};
        border: solid 1px ${palette.neutrals.borders};
        .btn-close {
          color: ${palette.fonts.primary};
          background-color: ${palette.secondary.neon[500]};
        }
        .toast-header {
          justify-content: space-between;
          background-color: ${palette.neutrals.primaryBg};
          button {
            margin-right: 0;
          }
          .header-title {
            font-weight: 600;
            width: 100%;
            color: ${palette.secondary.neon[500]};
          }
        }

        .body-content {
          width: 100%;
          display: flex;
          justify-content: space-between;
          .copied-value {
            color: ${palette.tertiary.pink[500]};
          }
        }
      }
    `;
  }};
`;
