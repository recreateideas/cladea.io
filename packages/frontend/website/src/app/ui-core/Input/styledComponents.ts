import styled from 'styled-components';
import { CustomCss } from '@types';

export const Container = styled.div<{ customCss?: CustomCss }>`
    width: 100%;
    display: flex;
    flex-direction: column;
    justify-content: center;
    ${(props) => props.customCss}
`;
