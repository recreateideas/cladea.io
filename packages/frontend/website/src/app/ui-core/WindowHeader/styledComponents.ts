import styled from 'styled-components';

export const Container = styled.div`
    -webkit-app-region: drag;
    cursor: default;
    display: flex;
`;

export const Button = styled.div`
    /* -webkit-app-region: no-drag; */
    font-size: 9pt;
    line-height: 11px;
    margin-left: 4px;
    margin-right: 3px;
    width: 11px;
    height: 11px;
    line-height: 10px;
    border-radius: 50%;
    display: inline-block;
    display: flex;
    justify-content: center;
    div {
        visibility: hidden;
    }
    color: black;
    &.min-btn {
        background: #ffbd4c;
        border: 1px solid #e09e3e;
        span {
            font-size: 19px;
        }
    }
    &.max-btn {
        background: #00ca56;
        border: 1px solid #00ca56;
        font-size: 10px;
    }
    &.close-btn {
        background: #ff5c5c;
        border: 1px solid #ff5c5c;
        span {
            font-size: 10px;
            top: 5px;
            position: absolute;
        }
    }
`;

export const SidebarSection = styled.div`
    -webkit-app-region: drag;
    min-width: 199px;
    background-color: ${(props) => props.theme.dsl.palette.primaryBg};
`;

export const PageSection = styled.div`
    width: 100%;
    -webkit-app-region: drag;
`;

export const DotsButtons = styled.div`
    -webkit-app-region: drag;
    display: flex;
    padding: 5px 3px 0px 3px;
    box-sizing: border-box;
    width: fit-content;
    &:hover div {
        visibility: visible;
    }
`;
