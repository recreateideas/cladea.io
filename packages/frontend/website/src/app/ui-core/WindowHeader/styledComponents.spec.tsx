import { SidebarSection } from './styledComponents';
import { TestProvider } from 'src/tests';

const { renderWithProviderAndTheme } = TestProvider({});

describe('styledComponents', () => {
    describe('<SidebarSection/>', () => {
        it('should render', () => {
            const root = renderWithProviderAndTheme(<SidebarSection />);
            expect(root).toMatchSnapshot();
        });
    });
});
