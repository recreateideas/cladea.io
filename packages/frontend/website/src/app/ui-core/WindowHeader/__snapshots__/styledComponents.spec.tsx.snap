// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`styledComponents <SidebarSection/> should render 1`] = `
.c0 {
  -webkit-app-region: drag;
  min-width: 199px;
}

<div
  class="c0"
/>
`;
